server/src/main/java/org/opensearch/telemetry/tracing/channels/TraceableTransportChannel.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.telemetry.tracing.channels;
+
+import org.opensearch.Version;
+import org.opensearch.common.util.FeatureFlags;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.transport.TransportResponse;
+import org.opensearch.telemetry.tracing.Span;
+import org.opensearch.telemetry.tracing.SpanScope;
+import org.opensearch.telemetry.tracing.Tracer;
+import org.opensearch.transport.TcpChannel;
+import org.opensearch.transport.TransportChannel;
+
+import java.io.IOException;
+
+/**
+ * Tracer wrapped {@link TransportChannel}
+ */
+public class TraceableTransportChannel implements TransportChannel {
+
+    private final TransportChannel delegate;
+    private final Span span;
+    private final Tracer tracer;
+
+    /**
+     * Constructor.
+     * @param delegate delegate
+     * @param span span
+     * @param tracer tracer
+     */
+    public TraceableTransportChannel(TransportChannel delegate, Span span, Tracer tracer, TcpChannel tcpChannel) {
```

Comment:
`TcpChannel tcpChannel` is not used

```suggestion
    public TraceableTransportChannel(TransportChannel delegate, Span span, Tracer tracer) {
```